src/Entities/Chunk.php

```diff
@@ -0,0 +1,386 @@
+    public function get_end(): string {
+        if ( ! $this->is_data_fetched ) {
+            $this->fetch_data();
+        }
+        return $this->end;
+    }
+
+    /**
+     * Get the duration in seconds
+     *
+     * @return float
+     */
+    public function get_duration(): float {
+        if ( ! $this->is_data_fetched ) {
+            $this->fetch_data();
+        }
+
+        // If duration is not yet calculated, calculate it
+        if ( empty( $this->duration ) && ! empty( $this->end ) && ! empty( $this->start ) ) {
+            return $this->calculate_duration();
+        }
+
+        return (float) $this->duration;
+    }
+
+    /**
+     * Get formatted duration in human-readable format
+     *
+     * @return string
+     */
+    public function get_formatted_duration(): string {
+        $duration = $this->get_duration();
+
+        if ( 0.0 === $duration ) {
+            return '0s';
+        }
+
+        // Format with microseconds precision
+        return sprintf( '%.6fs', $duration );
+    }
+
+    /**
+     * Calculate duration between start and end time
+     *
+     * @return float Duration in seconds with microseconds precision
+     */
+    private function calculate_duration(): float {
+        if ( empty( $this->end ) || empty( $this->start ) ) {
+            return 0.0;
+        }
+
+        $end_time   = (float) $this->end;
+        $start_time = (float) $this->start;
+
+        $duration = $end_time - $start_time;
+
+        // Store the calculated duration
+        $this->duration = (string) $duration;
+
+        return $duration;
+    }
+
+    /**
+     * Update multiple chunk properties at once
+     *
+     * @param array{
+     *     name?: string,
+     *     group?: string,
+     *     status?: string,
+     *     start?: string,
+     *     end?: string,
+     *     action_id?: int,
+     *     data?: array<mixed>
+     * } $arguments The update arguments
+     * @return void
+     */
+    public function update( array $arguments ): void {
+        $update_args = [];
+
+        // Handle data separately due to serialization
+        if ( isset( $arguments['data'] ) ) {
+            $this->data = $arguments['data'];
+            $update_args['data'] = serialize( $arguments['data'] );
+            unset( $arguments['data'] );
+        }
+
+        // Handle other fields
+        $allowed_fields = array(
+            'name'      => '%s',
+            'group'     => '%s',
+            'status'    => '%s',
+            'start'     => '%s',
+            'end'       => '%s',
+            'action_id' => '%d'
+        );
+        $allowed_fields = apply_filters( 'asp/chunks/allowed_fields', $allowed_fields );
+        $allowed_fields = array_keys( $allowed_fields );
+        foreach ( $allowed_fields as $field ) {
+            if ( isset( $arguments[ $field ] ) ) {
+                $this->{$field} = $arguments[ $field ];
+                $update_args[ $field ] = $arguments[ $field ];
+            }
+        }
```

Comment:
What is that about? Why can i add new fields in the first place?

Reply:
I just wanted to leave the possibility open for other plugins to append stuff to the database table. Maybe there is data missing in an edge case where I want to append it to the chunks. If this possibility is not anticipated, I can simplify the save methods.